Fix/linkta 321 remove firebase oauth flow from the related files

zhiyi15 opened the pull request:
<!--# Pull Request Template -->

## Description
Removed Oauth flow from firebase.ts and PrelaunchForm.tsx

<!--
Please include a summary of the change and which issue is fixed. Include relevant motivation and context. List any dependencies that are required for this change.

Linking Issues:
 - To associate this pull request with a GitHub issue within the same repository, use the "Fixes #IssueNumber" syntax.
 - For external issues, use Markdown links: [Issue Name](Issue Link).
-->

Fixes # (issue)321

## Type of change

<!--Please delete options that are not relevant.-->

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [x] Other 🗒️ **\*\***\_\_**\*\***

## How Has This Been Tested?

<!--Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Remove if not applicable. -->

Ran `npm run dev` to make sure everything else is working normally

## Checklist:

<!--Before submitting your pull request, please review the following checklist:-->

- [x] I have performed a self-review of my own code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have reviewed the [**Security Best Practices Document**](SECURITY_BEST_PRACTICES.md).
- [ ] I have followed the [**Naming Conventions Guide**](NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [x] I have ensured that my pull request title is descriptive.

## Additional Information:

<!--Any additional information, configuration, or data that might be necessary to reproduce the issue or feature.-->

zhiyi15 committed:
Merge branch 'dev' of https://github.com/Linkta-org/landing-page into dev

zhiyi15 committed:
Merge branch 'dev' of https://github.com/Linkta-org/landing-page into dev

zhiyi15 committed:
Merge branch 'dev' of https://github.com/Linkta-org/landing-page into dev

zhiyi15 committed:
removed google & GH provider signIn logic

zhiyi15 committed:
remove  oauth onclick features on Prelaunch signupForm

zhiyi15 committed:
removed createUserDoc for Oauth user; removed useEffect with oauth redirect logic in PreLaunchForm

zhiyi15 committed:
removed unused imports in firebase.ts and prelaunchform.tsx

zhiyi15 committed:
minor fix

zhiyi15 committed:
added back auth.useDeviceLanguage() for better user experience

kinskiwu commented on app/config/firebase.ts:
```diff
@@ -1,22 +1,10 @@
-interface ExtendedUser extends User {
-  reloadUserInfo?: ReloadUser;
-}
+import { getFirestore, doc, getDoc, setDoc, serverTimestamp } from 'firebase/firestore';
```
Since the db functionality now lives in the signup form I don't think you need to keep this import anymore.

zhiyi15 replied:
getFirestore should stay in this file because it is the entry point for all firebase service, but i will remove the rest of the imports.

kinskiwu commented on app/components/main-content/PrelaunchSignUpForm.tsx:
```diff
@@ -79,14 +63,8 @@ const PrelaunchSignUpForm: React.FC<PrelaunchSignUpFormProps> = ({ setFlowState
-
           <section aria-label="Sign Up with Email">
-            <h3>or sign up with email</h3>
+            <h3>Sign up with email</h3>
```
Since Darren is working on styling it may be better not to make any changes to the UI unless it is related to this PR.

zhiyi15 replied:
Certainly, i will change it back.

kinskiwu commented on app/components/main-content/PrelaunchSignUpForm.tsx:
```diff
@@ -31,21 +24,9 @@ const PrelaunchSignUpForm: React.FC<PrelaunchSignUpFormProps> = ({ setFlowState
 
   async function handleSubmit(values: FormValues) {
-    const { email, name } = values;
+    const { email, name, interests, source, features } = values;
```
Updates related to Firestore are implemented in #39, unless it is causing errors it may be better to leave it to the other ticket and not make changes in this PR.

zhiyi15 replied:
I actually made these changes before knowing about #39 PR. I thought it would be fine to leave it since this PR would probably be merge first before merging #39. But i will change it back to untouched.

kinskiwu replied:
Thanks!

kinskiwu commented on app/components/main-content/PrelaunchSignUpForm.tsx:
```diff
@@ -2,15 +2,8 @@
 import { TextInput, Button, Box } from '@mantine/core';
 import { useForm } from '@mantine/form';
-import { useEffect, Dispatch, SetStateAction, useMemo } from 'react';
+import { Dispatch, SetStateAction, useMemo } from 'react';
```
Are we using Dispatch and SetStateAction in this component?

zhiyi15 replied:
I'm not sure about Dispatch and SetStateAction as it wasn't related to Oauth flow. 
But i can see that Dispatch and SetStateAction are used here
`interface PrelaunchSignUpFormProps {
  setFlowState: Dispatch<SetStateAction<FlowState>>;
}`.

kinskiwu replied:
I see. Maybe better keep it there for now.

kinskiwu reviewed (changes requested):
Thanks for the good work! I left some comments for small changes.

zhiyi15 committed:
update changes based on change request on PR

dcpavel commented:
Have we tested that signing up via email still works after these changes are made?

ejwoods commented on app/components/main-content/PrelaunchSignUpForm.tsx:
```diff
@@ -79,12 +60,6 @@ const PrelaunchSignUpForm: React.FC<PrelaunchSignUpFormProps> = ({ setFlowState
-          </section>
-
           <section aria-label="Sign Up with Email">
             <h3>or sign up with email</h3>
```
I think this section and heading component can be removed now that we only have one sign up section.

zhiyi15 replied:
That’s weird, I thought I removed the buttons for Oauth already. Will definitely remove it

zhiyi15 replied:
Wait, I did remove the oauth section, is that what you meant?

ejwoods replied:
You removed the section with the OAuth buttons, but we can also remove the `<section>` and `<h3>` HTML elements for the "sign up with email" section. These elements were only needed to separate the OAuth and Email sections from one another - now that there's only one section, we don't need the elements distinguishing them.

ejwoods replied:
Ah, I see this was mentioned in a previous comment - https://github.com/Linkta-org/landing-page/pull/38#discussion_r1573476276

I would consider fixing this part of the UI related to this PR. But if this UI fix is not implemented here, we should make a ticket to ensure it gets done @kinskiwu

kinskiwu replied:
@dcpavel Would removing this section have a big impact on your styling? I think it is ok to remove the h3 but we may want to keep the section tag for styling.

kinskiwu replied:
@zhiyi15 I looked at Darren's ticket and it should be safe to remove h3: or sign up with email. Pls keep the section tag.

ejwoods reviewed (changes requested):
Looks good! I left one comment but ready to approve after that's resolved.

zhiyi15 commented:
> Have we tested that signing up via email still works after these changes are made?

Yes I have tested. Due to the current firestore security rule is set for only allow authenticated user to read and write into DB, so technically signing up via email is not working yet.

zhiyi15 committed:
added space

ejwoods committed:
remove h3 element from sign up with email section

Removed h3 HTML element containing text "or sign up with email" since this is now the only sign up section.

ejwoods reviewed (approved):
Removed the `<h3>` element I previously requested changes on but left the `<section>` element so as not to conflict with PR #23.
